base_layer/core/src/mining/miner.rs

```diff
@@ -172,49 +174,43 @@ impl<B: BlockchainBackend> Miner<B> {
     }
 
     // add the coinbase to the NewBlockTemplate
-    fn add_coinbase(&self, block: &mut NewBlockTemplate) -> Result<(), MinerError> {
+    fn add_coinbase(&self, block: &mut NewBlockTemplate) -> Result<(UnblindedOutput), MinerError> {
         let fees = block.body.get_total_fee();
         let height = block.header.height;
-        let (tx_id, key, r) = self.get_spending_key(
-            fees + self.consensus.emission_schedule().block_reward(height),
-            height + ConsensusConstants::current().coinbase_lock_height(),
-        )?;
+        let (key, r) = self.get_spending_key()?;
         let factories = CryptoFactories::default();
         let builder = CoinbaseBuilder::new(factories.clone());
         let builder = builder
             .with_block_height(block.header.height)
             .with_fees(fees)
             .with_nonce(r)
-            .with_spend_key(key);
+            .with_spend_key(key.clone());
         let tx = builder
             .build(self.consensus.clone())
             .expect("invalid constructed coinbase");
-        self.submit_tx_to_wallet(&tx, tx_id);
         block.body.add_output(tx.body.outputs()[0].clone());
         block.body.add_kernel(tx.body.kernels()[0].clone());
-        Ok(())
+        let unblinded_output = UnblindedOutput::new(
+            fees + self.consensus.emission_schedule().block_reward(height),
+            key,
+            Some(tx.body.outputs()[0].features.clone()),
+        );
```

Comment:
I'm guessing that almost every use case for `CoinbaseBuilder` will also want access to the `UnblindedOutput` data.

Why don't we have `CoinbaseBuilder.build()` return `Result<(Transaction, UnlbindedOutput), CoinbaseBuildError>`?

This way we get the key and output value from the horse's mouth and can't make any accidental mistakes down the road

Reply:
I agree, `CoinbaseBuilder.build()` now returns the unblinded_output as well